packages/components/button/src/ButtonGroup/ButtonGroup.tsx

```diff
@@ -0,0 +1,54 @@
+import { cx } from 'emotion';
+import React from 'react';
+import { CommonProps, Box } from '@contentful/f36-core';
+import getStyles from './ButtonGroup.styles';
+import type { ButtonGroupVariants, ButtonGroupSpacing } from './types';
+
+export interface ButtonGroupProps extends CommonProps {
+  /**
+   * Determines how the Button Group will display the buttons
+   * @default collapse
+   */
+  variant?: ButtonGroupVariants;
```

Comment:
I wonder if it's possible to type that `withDivider` exists only when variant is `collapse`, and `spacing` only when variant is `separate`? 

something like:

```
{ variant: 'collapse', withDivider?: boolean } | { variant: 'separate', spacing: Spacing }
```